From 19bd9a576434acd0640862bfc5db7e2bc4d3e70d Mon Sep 17 00:00:00 2001
From: alexandru Badicioiu <Alexandru.Badicioiu@freescale.com>
Date: Thu, 22 Mar 2012 21:48:51 +0000
Subject: [PATCH 126/518] Move loopback setting on DL_RX port from IPsec
 initialization to module initialization.

Signed-off-by: Alexandru BADICIOIU <alexandru.badicioiu@freescale.com>
[Grabbed from the branch, LINUX_IR5.2.0, of
https://git.freescale.com/git-private/cgit.cgi/ppc/alu-b4860/linux.git.]
Signed-off-by: Tiejun Chen <tiejun.chen@windriver.com>
---
 .../staging/fsl_dpa_offload/usecases/fm_utils.c    |    2 --
 .../staging/fsl_dpa_offload/usecases/fmc_config.c  |   16 ----------------
 drivers/staging/fsl_dpa_offload/usecases/main.c    |   15 +++++++++++++++
 3 files changed, 15 insertions(+), 18 deletions(-)

diff --git a/drivers/staging/fsl_dpa_offload/usecases/fm_utils.c b/drivers/staging/fsl_dpa_offload/usecases/fm_utils.c
index fe251fc..4fa5c8b 100644
--- a/drivers/staging/fsl_dpa_offload/usecases/fm_utils.c
+++ b/drivers/staging/fsl_dpa_offload/usecases/fm_utils.c
@@ -180,8 +180,6 @@ int config_loopback(int fm_id, int port_id)
 	iounmap((u32 *)((void *)(uintptr_t)
 			((addr + FM_1GMAC_CMD_CONF_CTRL_OFFSET))));
 
-	printk(KERN_INFO "Loopback was set on mac device\n");
-
 	return 0;
 
 }
diff --git a/drivers/staging/fsl_dpa_offload/usecases/fmc_config.c b/drivers/staging/fsl_dpa_offload/usecases/fmc_config.c
index f288841..7547ba3 100644
--- a/drivers/staging/fsl_dpa_offload/usecases/fmc_config.c
+++ b/drivers/staging/fsl_dpa_offload/usecases/fmc_config.c
@@ -70,12 +70,6 @@ static uint16_t mtu_post_enc;
 module_param(mtu_post_enc, ushort, 0);
 MODULE_PARM_DESC(mtu_post_enc, "\tMTU for OH port post encryption");
 
-/* Loopback operation mode */
-static int enable_loopback;
-module_param(enable_loopback, int, 0);
-MODULE_PARM_DESC(enable_loopback,
-			"\tEnable loopback for DownLink Interface\n");
-
 t_Handle h_CcNodeInRx;
 t_Handle h_CcNodeOutPreEnc[DPA_IPSEC_MAX_SUPPORTED_PROTOS];
 t_Handle h_CcNodeInPostDec;
@@ -1343,16 +1337,6 @@ int do_fmc_config(struct dpa_ipsec_params *dipParams)
 	if (!port || !port->pcd_owner_params.dev)
 		return E_NOT_AVAILABLE;
 
-	if (enable_loopback) {
-		error = config_loopback(FM, DL_RX);
-		if (error != E_OK) {
-			pr_err("%s : Cannot set loopback on interface!\n",
-				__func__);
-			return error;
-		}
-	}
-
-
 	pr_info("%s DL_RX\n", port->name);
 	ports[0] = &port->h_Dev;
 
diff --git a/drivers/staging/fsl_dpa_offload/usecases/main.c b/drivers/staging/fsl_dpa_offload/usecases/main.c
index a348218..64fd499 100644
--- a/drivers/staging/fsl_dpa_offload/usecases/main.c
+++ b/drivers/staging/fsl_dpa_offload/usecases/main.c
@@ -89,6 +89,12 @@ static int ipsec_offld;
 module_param(ipsec_offld, int, 0);
 MODULE_PARM_DESC(ipsec_offld, "\tInitialize IPsec offloading");
 
+/* Loopback operation mode */
+static int enable_loopback;
+module_param(enable_loopback, int, 0);
+MODULE_PARM_DESC(enable_loopback,
+			"\tEnable loopback for DownLink Interface\n");
+
 /*******************************************************************/
 
 int __init usecase_init(void)
@@ -171,6 +177,15 @@ int __init usecase_init(void)
 	pr_info("Created UL TX QUEUE queue fqid %d\n",
 		qman_fq_fqid(&ul_tx_fq));
 
+	if (enable_loopback) {
+		ret = config_loopback(FM, DL_RX);
+		if (ret != E_OK) {
+			pr_err("%s : Cannot set loopback on DL_RX port!\n",
+				__func__);
+			return 0;
+		}
+		pr_info("DL_RX port MAC loopback set\n");
+	}
 	if (ipsec_offld) {
 		ret = DipInit();
 		if (ret != E_OK) {
-- 
1.7.5.4

